crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -203,18 +215,32 @@ def provided_with_enabled?(feature, platform = nil, arch = nil, check_enabled =
             found = true
 
             r = new(platform, arch, repo)
+
             answer &&= r.available?
+            unless r.available?
+              repos[:missing_repos] ||= {}
+              repos[:missing_repos][r.arch.to_sym] ||= []
+              unless repos[:missing_repos][r.arch.to_sym].include?(r.name)
+                repos[:missing_repos][r.arch.to_sym].push(r.name)
+              end
+            end
 
             break unless check_enabled
             answer &&= r.active?
+            next if r.active?
```

Comment:
For the record, I would have gone with the `unless r.active?` way of coding it, for consistency with the code earlier. But matter of taste, I guess.